src/main/scala-2.11/org/apache/james/gatling/control/UserFeeder.scala

```diff
@@ -1,30 +1,68 @@
 package org.apache.james.gatling.control
 
-import java.util.UUID
-
 import io.gatling.core.Predef._
+import org.apache.james.gatling.utils.RandomStringGenerator
 
 import scala.concurrent.ExecutionContext.Implicits.global
-import scala.concurrent.Future
+import scala.concurrent.duration.Duration.Inf
+import scala.concurrent.{Await, Future}
+import scala.util.Success
 
 object UserFeeder {
 
-  def createUserFeeder(userCount: Int): Array[Map[String, String]] = {
-    val domain = Domain(UUID.randomUUID().toString)
+  val USERNAME: String = "username"
+  val PASSWORD: String = "password"
+
+  def createUserFeeder(userCount: Int): Array[Map[String, String]] =
+    toFeeder(
+      awaitInitialization(
+        createRegisteredUserFutures(userCount)))
+
+  def createUserFeederWithInboxAndOutbox(userCount: Int): Array[Map[String, String]] =
+    toFeeder(
+      awaitInitialization(
+        createRegisteredUserFutures(userCount)
+          .map(addSystemMailboxes(_))))
 
+  private def awaitInitialization(userFutures: List[Future[User]]) =
+    Await.result(
+      Future.sequence(userFutures),
+      Inf)
+
+  private def createRegisteredUserFutures(userCount: Int): List[Future[User]] = {
+    val domain = Domain()
     JamesWebAdministration.addDomain(domain).get
```

Comment:
does this work : `addDomain(domain).map(generateUsers(userCount, _).map(registerUser(_)))` ?